### ISSUE TITLE
With the option -i any not all received packets are printed

### ISSUE THREAD

**Issue by thedestroyer82** [the-tcpdump-group/libpcap#679]
With the option -i any not all the received ip packets are printed on the console. For example with execution tcpdump -n -i any port 80 the first SYN-Packet is not printed(but received), but the rest of the tcp handshake is printed.

**Comment by guyharris** [comment]
What happens with `tcpdump -n -i any` without a filter?

And what happens with `tcpdump -n port 80`?

I.e., is the filter causing the problem, or is it capturing on the "any" device causing the problem?

**Comment by thedestroyer82** [comment]
The -i any is the problem.

**Comment by guyharris** [comment]
I.e., you don't see the problem when capturing on an individual interface, no matter what the filter is, and you do see the problem when capturing on the any device, no matter what the filter is?

**Comment by thedestroyer82** [comment]
i see the problem only if i am capturing on the any device.

**Comment by guyharris** [comment]
So do you see it if you capture on the any device but don't specify a filter?

**Comment by thedestroyer82** [comment]
If a use a filter(e.g. port 80) with -i any then i will not see the first SYN packet.

**Comment by guyharris** [comment]
What happens if you ***DON'T*** use a filter on the any device?

**Comment by schischi** [comment]
I have the same issue and I found a fix.

How to reproduce:
  - libpcap 1.8.1 (latest stable release);
  - 'Recent' linux kernel (TPACKET_V3 enable);
  - Listen on the 'any' interface;
Here's some code to reproduce the issue: [pcap_any.c](https://gist.github.com/schischi/cfcef5e6fc1ce033a5c85f8d450e062f)

This code is working fine with libpcap from master.
I did a git bissect, and backporting [this diff](https://github.com/the-tcpdump-group/libpcap/commit/c3c41321429c045c826e019081b991aaea1618e6) to libpcap 1.8.1 fixes the issue.
